csrc/device_lower/analysis/predicate_elimination.cpp

```diff
@@ -188,17 +188,24 @@ class ProducerConsumerPairAnalyzer : public OptOutDispatch {
         BestEffortReplay::replayPasC(
             producer, consumer, /*consumer_compute_at_axis=*/-1, pairwise_map)
             .getReplay();
-    for (auto [c, p] : pairwise_map.mapConsumerToProducer()) {
-      // replayPasC skips mapping after the compute at position
-      c2p[c] = p;
-    }
     std::unordered_set<IterDomain*> consumer_index_ids;
     if (consumer->definition()->isA<MmaOp>()) {
       // NOTE: if consumer_index_ids is empty, it will be ignored. We only fill
       // it for MmaOp for now in order to limit our changes to the only op that
       // currently requires this analysis.
-      consumer_index_ids =
-          lower_utils::getIndexIDs(producer, consumer, &c2p).second;
+
+      // We flow from mapped IDs to the consumer's loop domain
+      std::vector<IterDomain*> mapped_ids;
+      auto root2logical = pairwise_map.mapConsumerToProducer();
+      for (IterDomain* id : consumer->getMaybeRootDomain()) {
+        if (root2logical.find(id) != root2logical.end()) {
+          mapped_ids.push_back(id);
+        }
+      }
+      // This set will omit loop IDs that are not mapped to the producer, such
+      // as N dimensions when the producer is an A operand without broadcasts.
+      consumer_index_ids = lower_utils::getIdsAlongPathBetween(
+          /*from=*/mapped_ids, /*to=*/consumer->getLoopDomain());
```

Comment:
I think what the below code is supposed to do is that we already know that the consumer is fine without predicates, but we also need to make sure that omitting predicates doesn't cause out-of-bound accesses with the producer. I think what we should really do here would be that starting from the actual loop domain (not just `consumer->getLoopDomain()` because they may be promoted) and see if there's any path from the actual loop domain to the allocation domain of the producer. If a specific ID of the loop domain is not involved in the path, that loop ID can be ignored as it shouldn't affect the indexing of the producer. 

If we'd take this direction for more than just MmaOp, I think we should add more thorough testing. I guess we would probably want to do this only for MmaOp for now to unblock the progress.